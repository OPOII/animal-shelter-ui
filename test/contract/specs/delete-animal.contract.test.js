import { provider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

describe('Given An Animal service', () => {
    describe('Get a request to delete an animal', () => {
        var animalName = "Rex"
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'delete animal',
                uponReceiving: 'a request to delete an animal',
                withRequest: {
                    method: 'DELETE',
                    path: `/animals/${animalName}`,
                },
                willRespondWith: {
                    status: 204
                }
            });
        });

        it("Then it should return the right data", async() =>{
            const response = await AnimalController.delete(animalName);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
});