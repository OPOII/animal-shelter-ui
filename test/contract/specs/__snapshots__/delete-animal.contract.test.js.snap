// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal service Get a request to delete an animal Then it should return the right data 1`] = `""`;
